### PR TITLE
Added build-scripts for 8 PHP packages

### PR DESCRIPTION
**Pull request by nailushapotnuru**
Added buildscripts for below packages:

1. doctrine-cache 1.10.0 & 1.10.2
2. doctrine-reflection 1.1.0 &1
3. parser 4.10.4
4. phpcomplex 2.0.0
5. theseer_tokenizer 1.2.0 & 1.2.1
6. zipstream-php 2.1.0
7. twig 1.42.4 & 1.42.5
8. composer 1.10.6, 1.10.19 & 1.10.20

### CODE REVIEW

**Comment by seth-priya** [comment]
@vedang-wartikar @Ryan-Dmello please review

File: t/twig/twig_1.42.4_1.42.5_rhel_8.4.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#! /bin/bash
+# -----------------------------------------------------------------------------
+#
+# Package       : twig/twig
+# Version       : 1.42.4, 1.42.5
```
Comment (Ryan-Dmello) [inline t/twig/twig_1.42.4_1.42.5_rhel_8.4.sh:5]: As per line 18, build script is created for version v1.42.4.  Please update accordingly

File: t/theeser_tokenizer/theseer_tokenizer_1.2.0_1.2.1_rhel_8.4.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#! /bin/bash
+# -----------------------------------------------------------------------------
+#
+# Package       : theseer/tokenizer
+# Version       : 1.2.0, 1.2.1
```
Comment (Ryan-Dmello) [inline t/theeser_tokenizer/theseer_tokenizer_1.2.0_1.2.1_rhel_8.4.sh:5]: As per line 18, build script is created for version 1.2.0. Please update accordingly

File: d/doctrine_cache/doctrine-cache_1.10.0_1.10.2_rhel_8.4.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# -----------------------------------------------------------------------------
+#
+# Package	: cache
+# Version	: 1.10.0, 1.10.2
```
Comment (Ryan-Dmello) [inline d/doctrine_cache/doctrine-cache_1.10.0_1.10.2_rhel_8.4.sh:4]: As per line 18, build script is created for version 1.10.0. Please update accordingly

File: d/doctrine__reflection/doctrine-reflection_1.1.0_1_rhel_8.4.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# -----------------------------------------------------------------------------
+#
+# Package	: Doctrine-Reflection 
+# Version	: 1, 1.1.0
```
Comment (Ryan-Dmello) [inline d/doctrine__reflection/doctrine-reflection_1.1.0_1_rhel_8.4.sh:4]: As per line 18, build script is created for version 1. Please update accordingly

File: c/composer/composer_1.10.6_1.10.19_1.10.20_rhel_8.4.sh [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# -----------------------------------------------------------------------------
+#
+# Package	: composer
+# Version	: 1.10.6, 1.10.19, 1.10.20
```
Comment (Ryan-Dmello) [inline c/composer/composer_1.10.6_1.10.19_1.10.20_rhel_8.4.sh:4]: As per line 18, build script is created for version 1.10.19. Please update accordingly

File: b/behat_minkextension/behat_minkextension_v2.3_v2.3.1_rhel_8.4.sh [hunk 6]
```diff
@@ -0,0 +1,52 @@
+#! /bin/bash
+# -----------------------------------------------------------------------------
+#
+# Package       : MinkExtension
+# Version       : v2.3, v2.3.1
```
Comment (Ryan-Dmello) [inline b/behat_minkextension/behat_minkextension_v2.3_v2.3.1_rhel_8.4.sh:5]: As per line 18, build script is created for version v2.3. Please update accordingly

File: b/behat_minkbrowserkitdriver/behat_mink-browserkit-driver_v1.3.3_v1.3.4_rhel_8.4.sh [hunk 7]
```diff
@@ -0,0 +1,51 @@
+#! /bin/bash
+# -----------------------------------------------------------------------------
+#
+# Package       : behat/mink-browserkit-driver
+# Version       : v1.3.3, v1.3.4
```
Comment (Ryan-Dmello) [inline b/behat_minkbrowserkitdriver/behat_mink-browserkit-driver_v1.3.3_v1.3.4_rhel_8.4.sh:5]: As per line 18, build script is created for version v1.3.3. Please update accordingly

**Review by Ryan-Dmello — CHANGES_REQUESTED** [review]
For all the attached scripts, just observed that phpunit binary is installed outside directory "$HOME_DIR/$PACKAGE_NAME", and then used as "vendor/bin/phpunit" for running tests, inside the directory "$HOME_DIR/$PACKAGE_NAME". As a result, I am assuming, the build scripts will fail. Please confirm if the build script are running successfully. 

Also, I noticed that in doc string section, "Version" field mentions multiple versions. However the actual script execution is written for a single version. Is it a intended change? If not please update it to version that will be build by the script. 

Other than above comments, the changes looks ok to me.

File: t/twig/twig_1.42.4_1.42.5_rhel_8.4.sh (hunk 1 above)
Comment (nailushapotnuru) [reply t/twig/twig_1.42.4_1.42.5_rhel_8.4.sh:5]: The script is tested and working fine on rhel 8.4. Also parameterized the version in line 18.

File: t/theeser_tokenizer/theseer_tokenizer_1.2.0_1.2.1_rhel_8.4.sh (hunk 2 above)
Comment (nailushapotnuru) [reply t/theeser_tokenizer/theseer_tokenizer_1.2.0_1.2.1_rhel_8.4.sh:5]: The script is tested and working fine on rhel 8.4. Also parameterized the version in line 18.

File: d/doctrine_cache/doctrine-cache_1.10.0_1.10.2_rhel_8.4.sh (hunk 3 above)
Comment (nailushapotnuru) [reply d/doctrine_cache/doctrine-cache_1.10.0_1.10.2_rhel_8.4.sh:5]: The script is tested and working fine on rhel 8.4. Also parameterized the version in line 18.

File: b/behat_minkbrowserkitdriver/behat_mink-browserkit-driver_v1.3.3_v1.3.4_rhel_8.4.sh (hunk 7 above)
Comment (nailushapotnuru) [reply b/behat_minkbrowserkitdriver/behat_mink-browserkit-driver_v1.3.3_v1.3.4_rhel_8.4.sh:5]: The script is tested and working fine on rhel 8.4. Also parameterized the version in line 18.

File: b/behat_minkextension/behat_minkextension_v2.3_v2.3.1_rhel_8.4.sh (hunk 6 above)
Comment (nailushapotnuru) [reply b/behat_minkextension/behat_minkextension_v2.3_v2.3.1_rhel_8.4.sh:5]: The script is tested and working fine on rhel 8.4. Also parameterized the version in line 18.

File: c/composer/composer_1.10.6_1.10.19_1.10.20_rhel_8.4.sh (hunk 5 above)
Comment (nailushapotnuru) [reply c/composer/composer_1.10.6_1.10.19_1.10.20_rhel_8.4.sh:5]: The script is tested and working fine on rhel 8.4. Also parameterized the version in line 18.

File: d/doctrine__reflection/doctrine-reflection_1.1.0_1_rhel_8.4.sh (hunk 4 above)
Comment (nailushapotnuru) [reply d/doctrine__reflection/doctrine-reflection_1.1.0_1_rhel_8.4.sh:5]: The script is tested and working fine on rhel 8.4. Also parameterized the version in line 18.

**Review by Ryan-Dmello — APPROVED** [review]
Latest changes looks ok to me

File: t/theeser_tokenizer/theseer_tokenizer_1.2.0_1.2.1_rhel_8.4.sh [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+HOME_DIR=`pwd`
+if ! git clone $PACKAGE_URL $PACKAGE_NAME; then
+        echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
+                echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
+fi
+cd $HOME_DIR/$PACKAGE_NAME
+git checkout $PACKAGE_VERSION
+if [ ! $PACKAGE_VERSION -eq 1.1.3 ];then
+	if ! ant install-tools && ant generate-autoloader; then
```
Comment (vedang-wartikar) [inline t/theeser_tokenizer/theseer_tokenizer_1.2.0_1.2.1_rhel_8.4.sh:37]: you will need to add the installation steps of **ant** as well. please refer https://gist.github.com/19317362/6a3cbf535426fb475d4ff19023ec9110

File: p/pear_archive-tar/pear_archive-tar_1.4.11_1.4.12_rhel_8.4.sh [hunk 9]
```diff
@@ -0,0 +1,59 @@
+! /bin/bash
```
Comment (vedang-wartikar) [inline p/pear_archive-tar/pear_archive-tar_1.4.11_1.4.12_rhel_8.4.sh:1]: the script is not executing. please remove this line

File: t/theeser_tokenizer/theseer_tokenizer_1.2.0_1.2.1_rhel_8.4.sh (hunk 8 above)
Comment (bahetiamit) [reply t/theeser_tokenizer/theseer_tokenizer_1.2.0_1.2.1_rhel_8.4.sh:37]: @vedang-wartikar "ant" is installed using yum. Refer line 23

File: p/pear_archive-tar/pear_archive-tar_1.4.11_1.4.12_rhel_8.4.sh [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+if ! pear package; then
+        echo "------------------$PACKAGE_NAME:install_fails-------------------------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_Fails"
+        exit 0
+else
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Install_Successful"
+fi
+cd $HOME_DIR/$PACKAGE_NAME
+if ! pear run-tests -r; then
```
Comment (vedang-wartikar) [inline p/pear_archive-tar/pear_archive-tar_1.4.11_1.4.12_rhel_8.4.sh:48]: you will need install php-pear using

**yum install php-pear -y**

File: p/pear_archive-tar/pear_archive-tar_1.4.11_1.4.12_rhel_8.4.sh [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+# Package       : pear/archive_tar
+# Version       : 1.4.11
+# Source repo   : https://github.com/pear/Archive_Tar
+# Tested on     : RHEL 8.4
+# Maintainer    : Amit Baheti <aramswar@in.ibm.com>
+#
+# Pre-requisite : pear package manager should be available on the system
+#                 as it needs user interaction for installation and cannot 
+#                 cannot be installed using script. Install as specified on
+#                 https://pear.php.net/manual/en/installation.getting.php
```
Comment (vedang-wartikar) [inline p/pear_archive-tar/pear_archive-tar_1.4.11_1.4.12_rhel_8.4.sh:13]: I am able to install pear using "yum install php-pear -y" on RHEL and run the tests successfully. Please check if this is possible.

**Review by vedang-wartikar — CHANGES_REQUESTED** [review]

File: p/pear_archive-tar/pear_archive-tar_1.4.11_1.4.12_rhel_8.4.sh (hunk 11 above)
Comment (nailushapotnuru) [reply p/pear_archive-tar/pear_archive-tar_1.4.11_1.4.12_rhel_8.4.sh]: Done

**Review by vedang-wartikar — CHANGES_REQUESTED** [review]
@seth-priya looks good to merge

**Review by vedang-wartikar — APPROVED** [review]

**Comment by seth-priya** [comment]
@nailushapotnuru  we don't specify multiple versions in script name - if it is going to / expected to work for multiple versions, just remove all of them and make it <package_name>_<disto_name>_<distro_version>.sh, that will make it more generic and anyone who wants to try it for a different version won't be discouraged from doing that

**Comment by nailushapotnuru** [comment]
> 

Done

**Pull request merged**